nntrainer/tensor/tensor_pool.cpp

```diff
@@ -207,56 +205,152 @@ void TensorPool::deallocate() {
   }
 }
 
+const std::vector<unsigned int> &
+TensorPool::getExecutionOrder(const std::string &name) {
+  return std::get<SourceDetails>(getSourceSpec(name).details).exec_order;
+}
+
 /**
  * @brief     Expand the lifespan of the tensor with the given name
  *
  */
-void TensorPool::expand_lifespan(const std::string &name,
-                                 TensorLifespan lifespan) {
-  if (name_map.find(name) == name_map.end())
-    throw std::invalid_argument("Requested tensor not found");
+TensorPool::RequestSpec &
+TensorPool::expandLifespan(const std::string &name,
+                           const std::vector<unsigned> &exec_order,
+                           TensorLifespan lifespan) {
+  auto &spec = getSourceSpec(name);
+  expandLifespan(spec, exec_order, lifespan);
+  return spec;
+}
 
-  int parent_spec_idx = name_map[name];
-  while (pool[parent_spec_idx].dependent == true)
-    parent_spec_idx = pool[parent_spec_idx].token;
+void TensorPool::expandLifespan(RequestSpec &spec,
+                                const std::vector<unsigned int> &exec_order,
+                                TensorLifespan lifespan) {
+  auto &details = std::get<SourceDetails>(spec.details);
+  NNTR_THROW_IF((details.lifespan != TensorLifespan::UNMANAGED &&
+                 lifespan == TensorLifespan::UNMANAGED),
+                std::invalid_argument)
+    << "Extending to lifespan to unmanaged is not possible for name: "
+    << spec.tensor->getName();
+
+  if (details.lifespan != TensorLifespan::UNMANAGED) {
+    /// update only if lifespan is unmanaged
+    details.lifespan =
+      enum_class_or<TensorLifespan>(details.lifespan, lifespan);
+  }
+  details.exec_order.insert(details.exec_order.end(), exec_order.begin(),
+                            exec_order.end());
```

Comment:
Just checking, it is all we have to do, set the lifespan and execution order? Is the memory pool handling rest of them?

Reply:
Yes, we only need those two variable :)